packages/graphback-schema/src/schema.ts

```diff
@@ -21,6 +40,60 @@ ${types.map((name: string) => `type ${name}Pagination {
 `
 }
 
+const scalars = ['ID', 'String', 'Boolean', 'Int', 'Float']
+
+const validateFields = (types: string[], fields: string[]): string[] => {  
+  return fields.filter((s: string) => {
+    const removedDefs = s.replace('[]','').replace('!','')
+    const splitType = removedDefs.split(': ')[1]
+    if(types.includes(splitType)) {
+      return false
+    }
+    if(scalars.includes(splitType)) {
+      return true
+    }
+
+    return false
+  })
+}
+
+const inputFields = (fields: string[]): string[] => {
+  return fields.filter((f: string) => !f.startsWith('id'))
+}
+
+const inputs = (defs: Definition[]): string => {
```

Comment:
I think I get what this code does but overall readability is quite bad here. 
We do a lot of type juggling and it can be very difficult to get that. 

We do lots of map objects here and mixing it with actual schema definitions. 
Do you thing we can detach that into 2 areas:

- code here can build context object we need 
- We will have a strongly typed context object that we expect in schema definitions
- schema definitions could be still in template language
- we validate output schema before sending success. 

We need more than just get things working. 
We need to figure out architecture and tools on top of graphql code generator that will work for our use cases.

If you think that this approach is clean and easy we can stay with that, just not really convinced that this is a good option in terms of unit testing this.

Reply:
Yes it would be better

Reply:
Can we explore that approach then?